File: src/main/java/com/github/cannor147/request/RequestService.java
package com.github.cannor147.request;

import com.github.cannor147.model.Color;
import com.github.cannor147.painter.PainterKt;
import com.github.cannor147.request.colorization.ColorizationTask;

import java.awt.image.BufferedImage;
import java.util.Collections;
import java.util.Optional;
import java.util.Queue;

public class RequestService {
    public BufferedImage handleRequest(Request request) {
        final BufferedImage image = request.getGeoMap().copyMap();
        perform(image, request.getTasks());

        Optional.ofNullable(request.getGeoMap().getBackground())
                .map(background -> PainterKt.findArea(background, Color.WHITE.getRgbColor()))
                .orElseGet(Collections::emptySet)
                .forEach(point -> PainterKt.fillPoint(image, point, Color.WHITE.getRgbColor()));
        return image;
    }

    private void perform(BufferedImage image, Queue<ColorizationTask> tasks) {
        tasks.forEach(task -> perform(image, task));
    }

    private void perform(BufferedImage image, ColorizationTask task) {
        if (task.getOnlyPixel()) {
            task.getPoints().forEach(point -> PainterKt.fillPoint(image, point, task.getRgbColor()));
        } else {
            task.getPoints().forEach(point -> PainterKt.fillArea(image, point, task.getRgbColor()));
        }
    }
}

File: src/main/java/com/github/cannor147/request/RequestService.kt
package com.github.cannor147.request

import com.github.cannor147.model.Color
import com.github.cannor147.painter.fillArea
import com.github.cannor147.painter.fillPoint
import com.github.cannor147.painter.findArea
import com.github.cannor147.request.colorization.ColorizationTask
import java.awt.Point
import java.awt.image.BufferedImage
import java.util.*
import java.util.function.Consumer

class RequestService {
    fun handleRequest(request: Request): BufferedImage {
        val image = request.geoMap.copyMap()
        perform(image, request.tasks)
        Optional.ofNullable(request.geoMap.background)
            .map { background: BufferedImage? ->
                findArea(
                    background!!, Color.WHITE.rgbColor)
            }
            .orElseGet { emptySet() }
            .forEach(Consumer { point: Point? -> fillPoint(image, point!!, Color.WHITE.rgbColor) })
        return image
    }

    private fun perform(image: BufferedImage, tasks: Queue<ColorizationTask>) {
        tasks.forEach(Consumer { task: ColorizationTask -> perform(image, task) })
    }

    private fun perform(image: BufferedImage, task: ColorizationTask) = when {
        task.onlyPixel -> task.points.asSequence().forEach { fillPoint(image, it, task.rgbColor) }
        else -> task.points.asSequence().forEach { fillArea(image, it, task.rgbColor) }
    }
}